### DISCUSSION TITLE
"wrapping" or shadow for tracing

### DISCUSSION THREAD

**Discussion by DynV** [ImageMagick/ImageMagick#6367] [Help] [answered]
### ImageMagick version

7.0.10-29

### Operating system, version and so on

Windows 10 Pro 22H2

### Description

There's a transparent image with some very dark tracing to be displayed over dark making it difficult to see the lines. Is there a command to either
1. wrap the traces in very pale, preferably with rounded ends or
2. get the traces a very pale shadow?
If things go as I'm hoping for, the vast majority of the resulting image would be transparent.

Thank you kindly for your help

**Comment by snibgo** [comment]
Please show a sample input.

Shadows can be any colour you want, such as white.

**Reply by DynV** [reply]
Attached is from https://www.cleanpng.com/png-line-art-black-circle-clip-art-curved-line-773690/
![kisspng-line-art-black-circle-clip-art-curved-line-5ac053e4de4d36 7195906015225538289106](https://github.com/ImageMagick/ImageMagick/assets/2823807/821d4f30-4086-4159-b5d9-06aa5d1bf6fd)

**Comment by snibgo** [comment]
I use a white shadow (Windows syntax):
```
magick ^
  darkCurve.png ^
  -background White ^
  ( +clone ^
    -shadow 80x3+0+0 ^
  ) ^
  +swap ^
  -background None -layers merge +repage ^
  out.png
```
![s](https://github.com/ImageMagick/ImageMagick/assets/33812368/cc02e039-e32a-4a40-ba3d-d6339bb4ca4f)

Does that do what you want?

**Reply by DynV** [reply]
I does although I'd prefer if it was solid "wrapping" the original trace. Is there a way to put that in the command prompt--without--having to create some script file?

**Comment by snibgo** [comment]
>  I'd prefer if it was solid "wrapping" the original trace.

I don't understand what that means.

You can change the `-shadow` parameters, for example:
```
magick ^
  darkCurve.png ^
  -background White ^
  ( +clone ^
    -shadow 1000x30+0+0 ^
  ) ^
  +swap ^
  -background None -layers merge +repage ^
  s2.png
```
![s2](https://github.com/ImageMagick/ImageMagick/assets/33812368/03d43c57-e115-4591-80c8-1e3d141f34ec)

**Reply by DynV** [reply]
Perhaps you missed:
> Is there a way to put that in the command prompt--without--having to create some script file?

**Comment by snibgo** [comment]
I didn't understand that either.

I have shown you one command. You can copy and paste it into the command prompt. No script file is needed.

If you want it on just one line, remove each ^ at the end of the lines, and join the lines up.

**Reply by DynV** [reply]
I wrapping I meant like
![wrapping_tracing-indexed](https://github.com/ImageMagick/ImageMagick/assets/2823807/2b4a9c7b-448f-4c4e-9a96-3ae16d697c88)

**Comment by snibgo** [comment, marked as answer]
For that, I suggest morphology:
```
magick ^
  darkCurve.png ^
  ( +clone ^
    -alpha extract ^
    -morphology dilate disk:5 ^
    ( +clone -fill White -colorize 100 ) ^
    +swap ^
    -alpha off -compose CopyOpacity -composite ^
  ) ^
  -compose DstOver -composite ^
  s3.png
```

![s3](https://github.com/ImageMagick/ImageMagick/assets/33812368/e6de71cf-b7dd-4e13-8aa2-7fe669e20317)

**Reply by DynV** [reply]
I dropped dilate disk to 2 and it was exactly what I wanted. Thanks again. 😀

Perhaps you want to mention the names you think better qualify what I wanted, what I called wrapping, so someone has better chance to find this post.

**Comment by snibgo** [comment]
The task is to draw a solid-colour boundary around an opaque object.
